src/pages/match.tsx

```diff
@@ -38,29 +45,39 @@ const MatchInternal: FC<Props> = ({ roomId, session }) => {
         void roomData.refetch();
       },
       onError(error) {
-        console.error(error);
+        showBoundary(error);
       },
       enabled: roomData.data && !roomData.data.arePointsCalculated,
     },
   );
 
-  if (roomData.isError) return <ErrorView />;
-  if (!roomData.data) return;
+  if (!roomData.data || roomData.isError) {
+    const error = roomData.error?.message ?? errorMessage.match.lostConnection;
+    console.error(error);
+    showBoundary(new Error(error));
+    return;
+  }
```

Comment:
Again, I think the old declarative way was more straightforward and fits better in the react way of doing things, also because this could trigger `showBoundary` multiple times since it's not inside a handler or a `useEffect` and that's a leak. I think the best is to just add params to the error view to specify the error type and render that there.